### ISSUE TITLE
lower accuracy of pretrained models?

### ISSUE THREAD

**Issue by HobbitLong** [pytorch/vision#907]
I evaluated the pretrained models on ImageNet validation set. But somehow I could not get the number reported here: 
https://pytorch.org/docs/stable/torchvision/models.html

Using error rates of ResNet18 and ResNet34 as examples:
mine  / reported
31.628 / 30.24 (resnet18)
27.688 / 26.70 (resnet34)

maybe 3 possibilities
(1) the official reported number is from testing set so that my comparison is apple to orange
(2) Maybe my data is corrupted. But I cannot redownload the ImageNet data to verify it, because of some reason.
(3) my code has a bug

Below is my code, could anyone help test it? Thanks!

```
from __future__ import print_function

import torch
import time
import torch.nn as nn
from torch.utils.data import DataLoader
from torchvision import datasets
from torchvision import transforms


class AverageMeter(object):
    """Computes and stores the average and current value"""
    def __init__(self):
        self.reset()

    def reset(self):
        self.val = 0
        self.avg = 0
        self.sum = 0
        self.count = 0

    def update(self, val, n=1):
        self.val = val
        self.sum += val * n
        self.count += n
        self.avg = self.sum / self.count


def accuracy(output, target, topk=(1,)):
    """Computes the accuracy over the k top predictions for the specified values of k"""
    with torch.no_grad():
        maxk = max(topk)
        batch_size = target.size(0)

        _, pred = output.topk(maxk, 1, True, True)
        pred = pred.t()
        correct = pred.eq(target.view(1, -1).expand_as(pred))

        res = []
        for k in topk:
            correct_k = correct[:k].view(-1).float().sum(0, keepdim=True)
            res.append(correct_k.mul_(100.0 / batch_size))
        return res


def validate(val_loader, model, criterion, print_freq=10):
    """One epoch validation"""
    batch_time = AverageMeter()
    losses = AverageMeter()
    top1 = AverageMeter()
    top5 = AverageMeter()

    # switch to evaluate mode
    model.eval()

    with torch.no_grad():
        end = time.time()
        for idx, (input, target) in enumerate(val_loader):

            input = input.float()
            if torch.cuda.is_available():
                input = input.cuda()
                target = target.cuda()

            # compute output
            output = model(input)
            loss = criterion(output, target)

            # measure accuracy and record loss
            acc1, acc5 = accuracy(output, target, topk=(1, 5))
            losses.update(loss.item(), input.size(0))
            top1.update(acc1.item(), input.size(0))
            top5.update(acc5.item(), input.size(0))

            # measure elapsed time
            batch_time.update(time.time() - end)
            end = time.time()

            if idx % print_freq == 0:
                print('Test: [{0}/{1}]\t'
                      'Time {batch_time.val:.3f} ({batch_time.avg:.3f})\t'
                      'Loss {loss.val:.4f} ({loss.avg:.4f})\t'
                      'Acc@1 {top1.val:.3f} ({top1.avg:.3f})\t'
                      'Acc@5 {top5.val:.3f} ({top5.avg:.3f})'.format(
                       idx, len(val_loader), batch_time=batch_time, loss=losses,
                       top1=top1, top5=top5))

        print(' * Acc@1 {top1.avg:.3f} Acc@5 {top5.avg:.3f}'
              .format(top1=top1, top5=top5))

    return top1.avg, losses.avg


def test():

    test_folder = './data/imagenet/val'

    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],
                                     std=[0.229, 0.224, 0.225])
    test_transform = transforms.Compose([
        transforms.Resize(256),
        transforms.CenterCrop(224),
        transforms.ToTensor(),
        normalize,
    ])

    test_set = datasets.ImageFolder(test_folder, transform=test_transform)
    test_loader = DataLoader(test_set,
                             batch_size=256,
                             shuffle=False,
                             num_workers=8,
                             pin_memory=True)

    import torchvision.models as models
    model = models.resnet18(pretrained=True)
    model = model.cuda()
    criterion = nn.CrossEntropyLoss()
    criterion = criterion.cuda()

    acc, loss = validate(test_loader, model, criterion)

    print('testing accuracy: ', acc)


if __name__ == '__main__':
    test()

```

**Comment by fmassa** [comment]
Hi,

I just run the reference classification script from https://github.com/pytorch/vision/blob/master/references/classification/train.py, changing the line where we define the model to instead use a pre-trained resnet18 model, and I obtained exactly the reported results
```
* Acc@1 69.758 Acc@5 89.078
```

I'd try running the aforementioned script, and if the results don't match, maybe re-download the dataset / model?

**Comment by fmassa** [comment]
Given that I managed to match results for resnet18, I'm closing this issue, but feel free to reopen it if it still doesn't work.

Note that I'm using a recent PyTorch nightly, and torchvision from master.

**Comment by HobbitLong** [comment]
> Given that I managed to match results for resnet18, I'm closing this issue, but feel free to reopen it if it still doesn't work.
> 
> Note that I'm using a recent PyTorch nightly, and torchvision from master.

Thanks!

My friend has helped me identify this problem. The data is corrupted!

**Comment by fmassa** [comment]
Sure, I'm evaluating it now, I'll get back to you once it's done.

**Comment by fmassa** [comment]
Here it is the result of evaluating on the training set
```
* Acc@1 69.062 Acc@5 87.274
```

**Comment by HobbitLong** [comment]
> Here it is the result of evaluating on the training set for resnet18
> 
> ```
> * Acc@1 69.062 Acc@5 87.274
> ```

Oh, wait, it's way bad than what I get. It's even slightly worse than what you get on the validation set? Acc@1 69.758 Acc@5 89.078

**Comment by HobbitLong** [comment]
> Here it is the result of evaluating on the training set for resnet18
> 
> ```
> * Acc@1 69.062 Acc@5 87.274
> ```

Seems your results on the training set is not that reasonable. But it might be too much to ask you to help check this back and forth. Thanks a lot for being helpful!
